### ISSUE TITLE
waterloggable blocks spawn waterlogged when generating underwater

### ISSUE THREAD

**Issue by ChloeIsNobody** [TheGrimsey/Stoneholm#18]
I have been using this mod for some time, and I have noticed that whenever part of an underground village generates underwater, all waterloggable blocks in the village become waterlogged, causing the village to be partially flooded.

Looking at previous issues, I see that this has been a problem in the past, but was supposedly fixed in version 1.3. However, I am using version 1.4.2 on Minecraft 1.18.1, and am still encountering this bug.

**Comment by CoolGuyGit** [comment]
Yes I have noticied this too on 1.18.2 on mod version 1.4.3 where some lanterns have generated waterlogged.
